### PR TITLE
drogon: init at 1.6.0

### PR DESCRIPTION
**Pull request by urlordjames**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by urlordjames** [comment]
I did some additional testing with some CMake projects and I've discovered that in order to use `find_package` you need to be in a shell that includes this package's dependencies too.  It looks like `DrogonConfig.cmake` calls `find_dependency`, but for some reason these the dependencies it's looking for are not visible to CMake unless you are in a shell that includes them.  Does anyone more knowledgeable than me know how to fix this?

## example
would be included in CMakeLists.txt
```
find_package(Drogon REQUIRED)
```

a shell.nix that works now
```
buildInputs = with pkgs; [
  drogon
  jsoncpp
  libuuid
  zlib
];
```

a shell.nix that I think should work
```
buildInputs = with pkgs; [
  drogon
];
```

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10650,6 +10650,12 @@
     githubId = 33242106;
     name = "Uri Baghin";
   };
+  urlordjames = {
+    email = "urlordjames@gmail.com";
+    github = "urlordjames";
+    githubId = 32751441;
+    name = "urlordjames";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:10658]: Please make this into a separate commit:

```
maintainers: add urlordjames
```

File: pkgs/development/libraries/drogon/no_cmake_test.patch [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+-  fi
+-else
+-  if [ ! -f "Debug\drogon_test.exe" ]; then
+-      echo "Failed to build drogon_test"
+-      exit -1
+-  fi
+-fi
+-
+ cd ../../
+ rm -rf drogon_test
```
Comment (jonringer) [inline pkgs/development/libraries/drogon/no_cmake_test.patch:41]: I feel this makes the file irrelevant...

Why keep it around?

File: pkgs/development/libraries/drogon/no_unused_parameter.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f4b3b25..200870e 100755
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -71,6 +71,7 @@ endif (BUILD_DROGON_SHARED)
+ 
+ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
+     target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Werror)
++    set(CMAKE_CXX_FLAGS -Wno-unused-parameter)
```
Comment (jonringer) [inline pkgs/development/libraries/drogon/no_unused_parameter.patch:9]: This isn't nix-specific, should probably be upstreamed

Comment (urlordjames) [reply pkgs/development/libraries/drogon/no_unused_parameter.patch:9]: I did some more experimenting and I discovered that this patch is not needed if you include `openssl`.  For now I'll just include it, but I'll see if I can get this fixed upstream so I can make including `openssl` override-able.

File: pkgs/development/libraries/drogon/no_cmake_test.patch (hunk 2 above)
Comment (urlordjames) [reply pkgs/development/libraries/drogon/no_cmake_test.patch:41]: How does this patch make `test.sh` irrelevant?  It still does integration testing and it runs `drogon_ctl` a few times with different arguments to make sure it works right.

Comment (jonringer) [reply pkgs/development/libraries/drogon/no_cmake_test.patch:41]: do you mind commenting  (in the nix expression) the intent of the patch?

Comment (urlordjames) [reply pkgs/development/libraries/drogon/no_cmake_test.patch:41]: No problem, do you find these more helpful?
